license/src/com/untangle/app/license/LicenseManagerImpl.java

```diff
@@ -1265,4 +1290,38 @@ private String getServerParams()
             "&numDevices=" + numDevices +
             "&version=" + uvmVersion;
     }
+
+    /**
+     * This hook is called when license settings change
+     */
+    private class LicenseChangeHookCallback implements HookCallback {
+
+        /**
+         * Constructor
+         */
+        LicenseChangeHookCallback(){}
+
+        /**
+         * Get the name of callback hook
+         *
+         * @return The name
+         */
+        public String getName() {
+            return "license-mananger-license-change-hook";
+        }
+
+        /**
+         * Callback function
+         *
+         * @param args
+         *        Callback arguments
+         */
+        public void callback(Object... args) {
+            UvmContextFactory.context().appManager().syncWithLicenses(); 
+            if (isRestricted()) {
```

Comment:
why is the global needed? can we not just call UvmContextFactory.context().licenseManager().isRestricted() directly?